cores/arduino/WString.cpp

```diff
@@ -492,7 +492,13 @@ int String::compareTo(const String &s) const
 		if (buffer && len > 0) return *(unsigned char *)buffer;
 		return 0;
 	}
-	return strcmp(buffer, s.buffer);
+	const char *p1 = buffer;
+	const char *p2 = s.buffer;
+
+	while (*p1 == *p2++)
+		if ('\0' == *p1++)
+			return 0;
+	return (*(const char *)p1 - *(const char *)(p2 - 1));
```

Comment:
​Agree.  I will back out the change for strcmp issue. Erik,  the issue was
found by Janet Y Scholar and assigned to me.

On Thu, May 26, 2016 at 4:18 PM, SidLeung <notifications@github.com> wrote:

> In cores/arduino/WString.cpp
> <https://github.com/01org/corelibs-arduino101/pull/173#discussion_r64837141>
> :
>
> > @@ -492,7 +492,13 @@ int String::compareTo(const String &s) const
> >  		if (buffer && len > 0) return *(unsigned char *)buffer;
> >  		return 0;
> >  	}
> > -	return strcmp(buffer, s.buffer);
> > +	const char *p1 = buffer;
> > +	const char *p2 = s.buffer;
> > +
> > +	while (*p1 == *p2++)
> > +		if ('\0' == *p1++)
> > +			return 0;
> > +	return (*(const char *)p1 - *(const char *)(p2 - 1));
>
> Agree with Calvin. It may require more work. However, for future
> referencing and categorizing, it is desirable to separate out issues into
> individual reports. It will also ensure the discussion, analysis of the
> issues will not be mixed together.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/01org/corelibs-arduino101/pull/173/files/7a2670a5058904d53a4d77dbf07abc0c9b32ba16..2c7f034ea880175c476e63077df6ff4b5ee0e88f#r64837141>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AMk_wVrRN7cCMq7gN0pzZpHdoDx4IILzks5qFioqgaJpZM4ITBp_>
> .
>